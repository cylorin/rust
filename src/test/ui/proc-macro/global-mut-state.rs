// force-host
// no-prefer-dynamic

#![crate_type = "proc-macro"]
#![allow(warnings)]

use std::cell::Cell;
use std::sync::atomic::AtomicBool;

static mut FOO: u8 = 0;
//~^ ERROR mutable global state in a proc-macro

static BAR: AtomicBool = AtomicBool::new(false);
//~^ ERROR mutable global state in a proc-macro

thread_local!(static BAZ: Cell<String> = Cell::new(String::new()));
//~^ ERROR mutable global state in a proc-macro

static FROZEN: &str = "snow";
