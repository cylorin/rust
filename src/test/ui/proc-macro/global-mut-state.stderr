error: mutable global state in a proc-macro
  --> $DIR/global-mut-state.rs:10:1
   |
LL | static mut FOO: u8 = 0;
   | ^^^^^^^^^^^^^^^^^^^^^^^

error: mutable global state in a proc-macro
  --> $DIR/global-mut-state.rs:13:1
   |
LL | static BAR: AtomicBool = AtomicBool::new(false);
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: mutable global state in a proc-macro
  --> $DIR/global-mut-state.rs:16:1
   |
LL | thread_local!(static BAZ: Cell<String> = Cell::new(String::new()));
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)

error: aborting due to 3 previous errors

